src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -709,6 +705,10 @@ public function link($method = 'get')
 
         $node = $this->getNode(0);
 
+        if (!$node instanceof \DOMElement) {
+            throw new \InvalidArgumentException(sprintf('The current node list should contain only DOMElement instances, "%s" found.', get_class($node)));
```

Comment:
What do you suggest on this? Because only the first node is used my suggestion would be to reword the Exception like this:

    throw new \InvalidArgumentException(sprintf('The first node should be DOMElement instance, "%s" found.', get_class($node)));

For the corresponding PHPDoc:

    @throws \InvalidArgumentException If the current node list is empty or the first node is not instance of DOMElement